README.md

```diff
@@ -8,16 +8,207 @@ Local.
 For more information, see [Getting started with Cloudant NoSQL DB](https://www.ng.bluemix.net/docs/services/Cloudant/index.html)
 
 The `loopback-connector-cloudant` module is the Cloudant connector for the Loopback framework.
+<!-- TOC -->
+
+- [loopback-connector-cloudant](#loopback-connector-cloudant)
+- [Getting Started](#getting-started)
+    - [Design](#design)
+        - [Careful partial update](#careful-partial-update)
+        - [Careful Frequent Modification](#careful-frequent-modification)
+        - [Conflict Control](#conflict-control)
+    - [Model](#model)
+        - [Map between model and document](#map-between-model-and-document)
+        - [Model-specific configuration](#model-specific-configuration)
+        - [_rev property](#_rev-property)
+- [Setup Cloudant Instance](#setup-cloudant-instance)
+- [Installation](#installation)
+- [Configuration](#configuration)
+    - [Generate Datasource](#generate-datasource)
+    - [Datasource Config(`url` vs `username` & `password`)](#datasource-configurl-vs-username--password)
+    - [Requests Plugin](#requests-plugin)
+    - [Example Usage](#example-usage)
+- [CRUD](#crud)
+    - [Update](#update)
+- [Migration](#migration)
+    - [autoupdate vs automigrate](#autoupdate-vs-automigrate)
+    - [isActual](#isactual)
+    - [Example Code](#example-code)
+- [Discovery](#discovery)
+- [Query](#query)
+- [Testing](#testing)
+- [More Info](#more-info)
+- [Feature backlog](#feature-backlog)
+    - [View](#view)
+    - [Index](#index)
+
+<!-- /TOC -->
+
+# Getting Started
+
+## Design
+Loopback tries best to fit its model to a specific database's design, while limited by the nature of database, it's not always possible to support all Loopback features perfectly, and user would like to be aware of some key features about Cloudant before they start to design a Cloudant model.
+
+### Careful partial update
```

Comment:
@dhmlau I feel 'careful partial update' would be more clear :p 
well tbh it's just a personal feeling and maybe it's good to have feedback from more people @strongloop/sq-lb-apex 
two header options here: 
- option1: 'Careful partial update'
- option2: 'Partial update'
from a user's perspective, which one would you prefer to see?